pkg/term/windows/ansi_reader.go

```diff
@@ -0,0 +1,218 @@
+// +build windows
+
+package windows
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"unsafe"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// ansiReader wraps a standard input file (e.g., os.Stdin) providing ANSI sequence translation.
+type ansiReader struct {
+	file     *os.File
+	fd       uintptr
+	buffer   []byte
+	cbBuffer int
+	command  []byte
+	// TODO(azlinux): Remove this and hard-code the string -- it is not going to change
+	escapeSequence []byte
+}
+
+func newAnsiReader(nFile int) *ansiReader {
+	file, fd := getStdFile(nFile)
+	return &ansiReader{
+		file:           file,
+		fd:             fd,
+		command:        make([]byte, 0, ANSI_MAX_CMD_LENGTH),
+		escapeSequence: []byte(KEY_ESC_CSI),
+	}
+}
+
+// Close closes the wrapped file.
+func (ar *ansiReader) Close() (err error) {
+	return ar.file.Close()
+}
+
+// Fd returns the file descriptor of the wrapped file.
+func (ar *ansiReader) Fd() uintptr {
+	return ar.fd
+}
+
+// Read reads up to len(p) bytes of translated input events into p.
+func (ar *ansiReader) Read(p []byte) (n int, err error) {
+	if len(p) == 0 {
+		return 0, nil
+	}
+
+	for ar.cbBuffer <= 0 {
+		events, err := readInputEvents(ar.fd, len(p))
+		if err != nil {
+			return 0, err
+		}
+		ar.buffer = make([]byte, int(unsafe.Sizeof(events[0]))*len(events))
+		ar.cbBuffer = copy(ar.buffer, translateKeyEvents(events, ar.escapeSequence))
+	}
+
+	n = copy(p, ar.buffer)
+	ar.cbBuffer -= n
+	return n, nil
+}
+
+// readInputEvents polls until at least one event is available.
+func readInputEvents(fd uintptr, maxBytes int) ([]INPUT_RECORD, error) {
+	// Determine the maximum number of records to retrieve
+	// -- Cast around the type system to obtain the size of a single INPUT_RECORD.
+	//    unsafe.Sizeof requires an expression vs. a type-reference; the casting
+	//    tricks the type system into believing it has such an expression.
+	recordSize := int(unsafe.Sizeof(*((*INPUT_RECORD)(unsafe.Pointer(&maxBytes)))))
```

Comment:
I'm not sure I like this. It is dereferencing an `*INPUT_RECORD`, from a pointer that actually is a `*int`, but `INPUT_RECORD` > `int`, so bad things could happen. Or not, but either way it's confusing.

I suggest you have a private empty INPUT_RECORD in the package scope that way you can simply call `int(unsafe.Sizeof(emptyInputRecord))`